### PR TITLE
chore(helm-chart): add support for v1 Ingress to the chart

### PR DESCRIPTION
**Pull request by jkirkham-ratehub**
- Add a flag `ingress.legacy` to toggle between the deprecated
 Ingress versions and v1.  This defaults to `false` so it will upgrade
 an existing Ingress by default.  Set `ingress.legacy` to `true` if
 the older version of the Ingress are required.  Note this is a
 potential breaking change for some.
 - Split the "standard" and "legacy" ingress templates into separate
 files -- using template conditions to only use one of them.
 - Increment the chart version to 0.2.0.
 - Update the README.md.gotmpl to document the potential breaking
 changes.
 - Regenerate the helm-docs for the new chart version.
 - Add a json schema for the Helm values.yaml -- helm will
 automatically use this to validate values passed to it on install/upgrade.

### CODE REVIEW

**Comment by sseide** [comment]
thanks a lot. and specially for the schema file.

Using the new ingress as default is ok by now - everyone running much older versions should be aware of it and adapt things as needed...

Other things/questions as code remarks above.

**Comment by jkirkham-ratehub** [comment]
> thanks a lot. and specially for the schema file.
> 
> Using the new ingress as default is ok by now - everyone running much older versions should be aware of it and adapt things as needed...
> 
> Other things/questions as code remarks above.

Any updates on this?  I haven't seen any review comments yet.

**Comment by sseide** [comment]
> Any updates on this? I haven't seen any review comments yet.

Good question - for me the review comments from myself on your code are visible and pending... They are right below your commit. These are only the first remarks.

Any ideas why you do not see them?

image

**Comment by jkirkham-ratehub** [comment]
> > Any updates on this? I haven't seen any review comments yet.
> 
> Good question - for me the review comments from myself on your code are visible and pending... They are right below your commit. These are only the first remarks.
> 
> Any ideas why you do not see them?
> 
> image

Your screenshot says `Pending`.  Perhaps you didn't submit your review?

**Review by sseide — CHANGES_REQUESTED** [review]

**Comment by sseide** [comment]
yes - you where right. submit was missing. Sorry for the delay.

**Comment by sseide** [comment]
i will check it this week when i have some time. Thanks so far!

File: k8s/helm-chart/redis-commander/README.md.gotmpl [hunk 1]
```diff
@@ -38,14 +38,16 @@ ingress:
   hosts:
     - host: redis-ui.example.com
       paths: ["/"]
-EOF
-
-# install helm chart with the custom values
```
[the remark this answers is not in the document]
Comment (jkirkham-ratehub) [reply k8s/helm-chart/redis-commander/README.md.gotmpl]: Yeah, done.  Thanks.  Not sure how those got deleted.

File: k8s/helm-chart/redis-commander/README.md [hunk 2]
```diff
@@ -24,16 +24,18 @@ helm -n myspace install redis-web-ui ./k8s/helm-chart/redis-commander --set redi
 | fullnameOverride | string | `""` |  |
 | httpAuth.password | string | `""` | Specify http basic password for the web ui |
 | httpAuth.username | string | `""` | Specify http basic username and password to protect access to redis commander web ui |
-| image.apparmorProfile | string | `"runtime/default"` | Enable AppArmor per default when available on k8s host, change to "unconfined" to disable -- either AppArmor or SecComp may be enabled by the container runtime |
+| image.apparmorProfile | string | `"runtime/default"` | either AppArmor or SecComp may be enabled by the container runtime |
 | image.pullPolicy | string | `"Always"` | Deployment pull policy, either "Always" or "IfNotPresent" |
 | image.repository | string | `"rediscommander/redis-commander"` | Docker image for deployment |
-| image.seccompProfile | string | `"runtime/default"` | Enable SecComp profile when used by cluster, change to "unconfined" to disable -- either AppArmor or SecComp may be enabled by the container runtime |
```
[the remark this answers is not in the document]
Comment (jkirkham-ratehub) [reply k8s/helm-chart/redis-commander/README.md:30]: I'm using the latest helm-docs v1.7.0 and it does not merge consecutive comment blocks:  that means we only need one line with the `# --` to start a block and consecutive lines up until the setting (key) are used to render the associated comment.

**Comment by sseide** [comment]
Many thanks - merged code within f9e93edf and updated helm doc validation to use helm-docs 1.7.0 there too.

**Comment by sseide** [comment]
now i must find time to auto-publish this chart into a helm repo on local github page i think...

**Comment by jkirkham-ratehub** [comment]
Thanks for the support.  I was glad to help out.

**Pull request merged**